api/src/main/java/io/strimzi/api/kafka/model/status/Credential.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2019, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.status;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.strimzi.api.kafka.model.UnknownPropertyPreserving;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+
+/**
+ * Represents a single listener
+ */
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({ "credentialsSecret", "credentials" })
+@EqualsAndHashCode
+public class Credential implements UnknownPropertyPreserving, Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private String credentialsSecret;
+    private String credentials;
```

Comment:
I do not think the name `credentials` really describes the purpose. We should rethink it. Also, it might be multiple fields - e.g. `user.crt` and `user.key`.

Reply:
Whatever we do need to work for scramsha passwords too.

Reply:
Yes, but that should be easy since that is just one field.